main.php

```diff
@@ -51,23 +51,23 @@
         }
 
         // use defaults or force overwrite values
-        if (!isset($manifest["delimiter"]) || isset($config["parameters"]["delimiter"])) {
+        if (!isset($manifest["delimiter"]) || !empty($parameters["delimiter"])) {
```

Comment:
no jsem z toho trosku zmetenej, proc je to jednou isset a jednou empty? to znamena, ze nepujde nastavit prazdny enclosure, ne?

ten or znamena, ze parameters pretezuji manifest, ale kdyz to nebude nastaveny ani v jednom? to asi nejde, protoze je to prohnany pres ConfigDefinition

Reply:
tyjo, díky za komentář, došlo mi, proč to tam takhle bylo

představení
 - (A) `$config["parameters"]` - parametry v config.json
 - (B) `$parameters` - parametry z config.json prohnané přes ConfigDefinition
 - (C) `$manifest` - parametry v manifestu

```
if (!isset($manifest["enclosure"]) || isset($config["parameters"]["enclosure"])) {
  $manifest["enclosure"] = $parameters["enclosure"];
}
```

říká
- pokud 
   - není nastavený `enclosure` v manifestu nebo (C)
   - uživatel v parametrech procesoru nastavil `enclosure` (A)
 - tak
   - použij parametr prohnaný `ConfigDefinition` (hodota z A nebo default) (B)

jedině tak totiž dokážeš v manifestu přepsat nějakou už existující hodnotu na prázdnou (smazat primární klíč například)

jestli uživatel nastavil hodnotu v parametrech procesoru už v parametrech prohnaných `ConfigDefinition` nejde zjistit, tak už budou vždycky defaulty

Reply:
takže to `isset($config["parameters"][*])` tam bylo odůvodněně a naprosto chápu, že se to blbě čte. jak takovu věc napsat tak, aby to bylo srozumitelný?

Reply:
no ja bych to osobne nechal jak to bylo a dal k tomu link na tenhle komentar 
ale treba neco napadne @tomasfejfar  ;)

Reply:
Co takhle? Neextrahoval jsem to do úplně vlastních proměnných, zkouším to přes `array_keys` nejdřív, jestli to nebude srozumitelnější.

Reply:
Za mě určitě lepší. Ale je možné, že už moc chápu, co to dělá.